extended/src/main/java/io/kubernetes/client/extended/leaderelection/LeaderElector.java

```diff
@@ -97,14 +132,15 @@ private void renewLoop() {
     Duration retryPeriod = config.getRetryPeriod();
     long retryPeriodMillis = retryPeriod.toMillis();
     AtomicBoolean renewed = new AtomicBoolean(true);
+    long renewDeadlineMillis = config.getRenewDeadline().toMillis();
 
     ScheduledFuture scheduledFuture =
         scheduledWorkers.scheduleAtFixedRate(
             () -> {
               Future<Boolean> future = leaseWorkers.submit(this::tryAcquireOrRenew);
               try {
                 //  set renew success
-                renewed.set(future.get(retryPeriodMillis, TimeUnit.MILLISECONDS));
+                renewed.set(future.get(renewDeadlineMillis, TimeUnit.MILLISECONDS));
```

Comment:
I don't think this is what `renewDeadline` means. We should retry `tryAcquireOrRenew` even it return false within `renewDeadline` Duration. Just as what `client-go` does 
```go
wait.Until(func() {
		timeoutCtx, timeoutCancel := context.WithTimeout(ctx, le.config.RenewDeadline)
		defer timeoutCancel()
		err := wait.PollImmediateUntil(le.config.RetryPeriod, func() (bool, error) {
			done := make(chan bool, 1)
			go func() {
				defer close(done)
				done <- le.tryAcquireOrRenew()
			}()

			select {
			case <-timeoutCtx.Done():
				return false, fmt.Errorf("failed to tryAcquireOrRenew %s", timeoutCtx.Err())
			case result := <-done:
				return result, nil
			}
		}, timeoutCtx.Done())

		le.maybeReportTransition()
		desc := le.config.Lock.Describe()
		if err == nil {
			klog.V(5).Infof("successfully renewed lease %v", desc)
			return
		}
		le.config.Lock.RecordEvent("stopped leading")
		le.metrics.leaderOff(le.config.Name)
		klog.Infof("failed to renew lease %v: %v", desc, err)
		cancel()
	}, le.config.RetryPeriod, ctx.Done())
```